src/lib/comms/sol-network-impl-linux.c

```diff
@@ -91,12 +91,22 @@ static struct sol_network *network = NULL;
 
 SOL_API const char *
 sol_network_addr_to_str(const struct sol_network_link_addr *addr,
-    char *buf, uint32_t len)
+    struct sol_buffer *buf)
 {
+    const char *r;
+    size_t len;
+
     SOL_NULL_CHECK(addr, NULL);
     SOL_NULL_CHECK(buf, NULL);
+    len = buf->capacity - buf->used;
+    SOL_INT_CHECK(len, == 0, NULL);
+
+    r = inet_ntop(sol_network_sol_to_af(addr->family), &addr->addr,
+        sol_buffer_at_end(buf), len);
```

Comment:
hum, not good. This way you work with pre-allocated/static buffers, but you don't for empty buffers that would allocate memory. The ideal solution is to check if the buffer can be resized and if so, grow it by some threshold (ie: 32bytes), try it and handle `ENOSPC` in a loop. You can start the resize with a reasonable size, AFAIU we can know the size based on the family, right?